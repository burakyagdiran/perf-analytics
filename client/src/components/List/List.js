import './List.scss';

import React from 'react';
import PropTypes from 'prop-types';
import { Table } from 'antd';

function List({ data, ...props }) {
  const columns = [
    {
      title: 'Projects',
      dataIndex: 'hostName',
      key: 'hostName',
    },
  ];
  return (
    <div className="list-container">
      <Table
        rowKey="_id"
        className="table"
        columns={columns}
        dataSource={data}
        pagination={false}
      />
    </div>
  );
}

List.propTypes = {
  data: PropTypes.array.isRequired,
};

List.defaultProps = {
  data: [],
};

export default List;
