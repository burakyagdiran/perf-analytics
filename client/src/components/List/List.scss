@import '../../variables/colors.scss';

.list-container {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  margin-top: 20px;
  @media only screen and (min-width: 900px) {
    width: 650px;
    margin-left: 10px;
  }
  @media only screen and (max-width: 900px) {
    width: 500px;
  }
  @media only screen and (max-width: 700px) {
    width: 300px;
  }
  .table {
    width: 100%;
  }
}
