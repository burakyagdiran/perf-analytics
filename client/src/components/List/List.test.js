import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

it('should render correctly with no props', () => {
  const component = shallow(<List />);
  expect(component).toMatchSnapshot();
});

it('should render banner text correctly with given strings', () => {
  const testData = [];
  const component = shallow(<List title={testData} />);
  expect(component).toMatchSnapshot();
});

describe('List', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<List debug />);
    expect(component).toMatchSnapshot();
  });
});
