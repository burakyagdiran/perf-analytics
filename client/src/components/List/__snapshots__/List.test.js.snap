// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render correctly in "debug" mode 1`] = `
<div
  className="list-container"
>
  <Table
    className="table"
    columns={
      Array [
        Object {
          "dataIndex": "hostName",
          "key": "hostName",
          "title": "Projects",
        },
      ]
    }
    dataSource={Array []}
    pagination={false}
    rowKey="_id"
  />
</div>
`;

exports[`should render banner text correctly with given strings 1`] = `
<div
  className="list-container"
>
  <Table
    className="table"
    columns={
      Array [
        Object {
          "dataIndex": "hostName",
          "key": "hostName",
          "title": "Projects",
        },
      ]
    }
    dataSource={Array []}
    pagination={false}
    rowKey="_id"
  />
</div>
`;

exports[`should render correctly with no props 1`] = `
<div
  className="list-container"
>
  <Table
    className="table"
    columns={
      Array [
        Object {
          "dataIndex": "hostName",
          "key": "hostName",
          "title": "Projects",
        },
      ]
    }
    dataSource={Array []}
    pagination={false}
    rowKey="_id"
  />
</div>
`;
